secrets/alibaba/alibaba_ssm.go

```diff
@@ -0,0 +1,224 @@
+	getSecretParameterRequest := &oos20190601.GetSecretParameterRequest{
+		RegionId:       tea.String(a.region), //eu-central-1
+		Name:           tea.String(a.constructSecretPath(name)),
+		WithDecryption: tea.Bool(true),
+	}
+	runtime := &util.RuntimeOptions{}
+	retVal, tryErr := func() (_b []byte, _e error) {
+		defer func() {
+			if r := tea.Recover(recover()); r != nil {
+				_b = nil
+				_e = r
+			}
+		}()
+
+		response, err := a.client.GetSecretParameterWithOptions(getSecretParameterRequest, runtime)
+		if err != nil {
+			return nil, err
+		}
+
+		return []byte(tea.StringValue(response.Body.Parameter.Value)), nil
+	}()
+
+	if tryErr != nil {
+		a.logError(tryErr)
+	}
+
+	return retVal, tryErr
+}
+
+// SetSecret saves a secret to Alibaba SSM
+func (a *AlibabaSsmManager) SetSecret(name string, value []byte) error {
+	createSecretParameterRequest := &oos20190601.CreateSecretParameterRequest{
+		RegionId: tea.String(a.region), //eu-central-1
+		Name:     tea.String(a.constructSecretPath(name)),
+		Value:    tea.String(string(value)),
+	}
+	runtime := &util.RuntimeOptions{}
+	tryErr := func() (_e error) {
+		defer func() {
+			if r := tea.Recover(recover()); r != nil {
+				_e = r
+			}
+		}()
+
+		_, err := a.client.CreateSecretParameterWithOptions(createSecretParameterRequest, runtime)
+		if err != nil {
+			return err
+		}
+
+		return nil
+	}()
+
+	if tryErr != nil {
+		a.logError(tryErr)
+	}
+
+	return tryErr
+}
+
+// HasSecret checks if the secret is present on Alibabab SSM ParameterStore
+func (a *AlibabaSsmManager) HasSecret(name string) bool {
+	_, err := a.GetSecret(name)
+
+	return err == nil
+}
+
+// RemoveSecret removes a secret from Alibaba SSM ParameterStore
+func (a *AlibabaSsmManager) RemoveSecret(name string) error {
+	deleteSecretParameterRequest := &oos20190601.DeleteSecretParameterRequest{
+		RegionId: tea.String(a.region),
+		Name:     tea.String(a.constructSecretPath(name)),
+	}
+	runtime := &util.RuntimeOptions{}
+	tryErr := func() (_e error) {
+		defer func() {
+			if r := tea.Recover(recover()); r != nil {
+				_e = r
+			}
+		}()
+
+		_, err := a.client.DeleteSecretParameterWithOptions(deleteSecretParameterRequest, runtime)
+		if err != nil {
+			return err
+		}
+
+		return nil
+	}()
+
+	if tryErr != nil {
+		a.logError(tryErr)
+	}
+
+	return tryErr
+}
+
+func (a *AlibabaSsmManager) logError(err error) {
+	var e = &tea.SDKError{}
+	if _t, ok := err.(*tea.SDKError); ok { //nolint:errorlint
+		e = _t
+	} else {
+		e.Message = tea.String(err.Error())
+	}
+
+	_, err = util.AssertAsString(e.Message)
+	if err != nil {
+		a.logger.Error("unable to log error message")
+
+		return
+	}
+
+	a.logger.Error(tea.StringValue(e.Message))
+
+	var data interface{}
+
+	d := json.NewDecoder(strings.NewReader(tea.StringValue(e.Data)))
+
+	err = d.Decode(&data)
+	if err != nil {
+		a.logger.Error("unable to decode recommendation", err)
+
+		return
+	}
+
+	if m, ok := data.(map[string]interface{}); ok {
+		recommend, _ := m["Recommend"]
```

Comment:
Nitpick, either check if it exists in the map, or remove the second return value (`boolean`)

```suggestion
		recommend := m["Recommend"]
```